types/assert-plus/tslint.json

```diff
@@ -1,6 +1,7 @@
 {
     "extends": "@definitelytyped/dtslint/dt.json",
     "rules": {
+        "ban-types": false,
```

Comment:
those exclusion should be really thought about, those are not allowed, unless specific reason.
For this use case, you can remove this exclusion and add this in declaration file:
```ts
type Func = (...args: any[]) => any;
export {};
```
replacing every added `Function` with `Func` type . that will preserve declaration and removes error from linter.